mne_icalabel/label_components.py

```diff
@@ -30,10 +32,38 @@ def label_components(inst: Union[BaseRaw, BaseEpochs], ica: ICA, method: str):
 
     Returns
     -------
-    labels : np.ndarray of shape (n_components,) or (n_components, n_class)
-        The estimated numerical labels of each ICA component.
+    component_dict : dict
+        A dictionary with the following output:
+        - 'y_pred_proba' : np.ndarray of shape (n_components, n_classes)
+        Estimated corresponding predicted probabilities of output classes
+        for each independent component.
+        - 'y_pred' : list of shape (n_components,)
+        The corresponding numerical label of the class with the highest
+        predicted probability.
+        - 'labels': list of shape (n_components,)
+        The corresponding string label of each class in 'y_pred'.
+
+    Notes
+    -----
+    For ICLabel model, the output classes are ordered:
+    - 'Brain'
+    - 'Muscle'
+    - 'Eye'
+    - 'Heart'
+    - 'Line Noise'
+    - 'Channel Noise'
+    - 'Other'
     """
     _validate_type(method, str, "method")
     _check_option("method", method, methods)
     _validate_inst_and_ica(inst, ica)
-    return methods[method](inst, ica)
+    labels_pred_proba = methods[method](inst, ica)
+    labels_pred = np.argmax(labels_pred_proba, axis=1)
+    labels = [ICLABEL_NUMERICAL_TO_STRING[label] for label in labels_pred]
+
+    component_dict = {
+        "y_pred_proba": labels_pred_proba,
+        "y_pred": labels_pred,
+        "labels": labels,
+    }
+    return component_dict
```

Comment:
Any reason for nesting the output in a dictionary? Can't we directly have `return y_pred_proba, y_pred, labels` (in this order or another)?

Reply:
No real preference.

I only did this to make it super explicit and limit the number of return variables / make it a one-liner to convert to a Dataframe. Would you prefer the 3 tuple return instead?

Reply:
No strong preference either, but you can't convert this to a DataFrame that easily either since `labels_pred_proba` is a 2D array.

Reply:
Ah fair. Hmm now that I think about it more I do prefer in general single return arguments, but I'm also unsure if this is optimal here... I'll leave this open for now and we can always change it.

Reply:
I like the idea of simple conversion to DataFrame for the user (of course not on our end, let's not have pandas as a dependency just for that). With that in mind, I see 2 individual DataFrame in the proposed output:

| idx | labels | y_pred |
|-----|-------|--------|
| 0   | Eye   | 0.89   |
| 1   | Brain | 0.67   |
| ... | ...   | ...    |

and:

| idx    | Brain | Muscle | Eye  | Heart | Line Noise | Channel Noise | Other |
|--------|-------|--------|------|-------|------------|---------------|-------|
| 0      | 0.2   | 0.89   | 0.1  | 0.05  | 0.03       | 0.07          | 0.08  |
| 1      | 0.03  | 0.67   | 0.06 | 0.04  | 0.1        | 0.3           | 0.09  |
| ...    | ...   | ...    | ...  | ...   | ...        | ...           | ...   |
| n_comp | 0.3   | 0.23   | 0.43 | 0.22  | 0.11       | 0.08          | 0.7   |

So how about 2 outputs: `return dict(labels=labels, y_pred=labels_pred), {value: labels_pred_proba[:, k] for k, value in ICLABEL_NUMERICAL_TO_STRING.items()}`.

IMO, for what needs to be stored in the mne ICA instance, we do not need the `y_pred`, but only the `labels` and the second dict.

With this solution, you can get the 'raw' numpy array of probabilities with `mne_icalabel.iclabel.iclabel_label_compoments` or dictionaries ready to be transformed to DataFrames with  `mne_icalabel.label_components`.

WDYT?